### ISSUE TITLE
Inferring playback intervals in the absence of an initial mode martyria

### ISSUE THREAD

**Issue by Beatl3** [neanes/neanes#1164]
Again, in the context of typesetting a manuscript I do not wish to have the typical mode martyrias in the beginning of every hymn. 

That of course messes up the intervals in playback for the portion of the score that is the beginning of the hymn to the first martyria (which I set manually). 

I would like if possible to have the system infer what the intervals should be for the initial part of the hymn by "backward inference" from the first martyria. 

Thanks for considering!

**Comment by danielgarthur** [comment]
I don't think it would be possible to infer correctly. For example, in the first mode, hymns often end on PA, but the next hymn starts on DI. In this case the hymns end with a DI martyria, even though the last note was PA. Attempting to infer backwards from the DI martyria would result in incorrect playback.

I've occasionally considered allowing a way to turn any object into a "mode key". For example, you could click a text box or even a neume and indicate that the note or scale should change. Maybe something like that would be better.

**Comment by Beatl3** [comment]
I was thinking about inferring backwards from the first martyria occurring in the course of the musical line.

Given that the start of a new hymn or musical unit starts with a large capital letter, one could infer backwards from the first (manually set) occurring martyria in the course of that hymn to the start of the hymn. Does it make sense?

**Comment by danielgarthur** [comment]
I don't think it is safe to assume that the hymn always starts with a drop cap. I think that's especially not true for Arabic RTL settings. I have seen documents made (although not with Neanes) that did not include the drop caps in all the usual placces, probably in the interest of saving space or ink since the documents were made to be printed for use at the chanters stand instead of compiled into a book.

**Comment by gugushvili** [comment]
A possible solution could be to attach an extra menu to the martyria flushed to the right, which may offer an option of choosing the mode. This is a lot of extra work, of course.

Typing Mode 4 Resurrectional kathismata drove me crazy.

**Comment by danielgarthur** [comment]
I think this was recently resolved with the addition of custom mode keys. See the docs [here](https://neanes.github.io/neanes/guide/advanced.html#custom-mode-keys) for more info.
